Ginger/Ginger/Actions/ActionEditPages/ActExcelEditPage.xaml.cs

```diff
@@ -96,21 +93,62 @@ private void BrowseExcelButton_Click(object sender, RoutedEventArgs e)
             }) is string fileName)
             {
                 ExcelFileNameTextBox.ValueTextBox.Text = fileName;
+                SheetsList = null;
+                // Empty the list in the UI before Creating a new list
+                GingerCore.General.FillComboFromList(SheetNamComboBox, SheetsList);
                 FillSheetCombo();
             }
         }
 
-        private void FillSheetCombo()
+        private async Task FillSheetCombo(bool initialLoad = false )
         {
             ContextProcessInputValueForDriver();
-            //Move code to ExcelFunction no in Act...
-            List<string> SheetsList = mExcelOperations.GetSheets(mAct.CalculatedFileName);
-            if (SheetsList == null || SheetsList.Count == 0)
+            if (SheetsList == null || !SheetsList.Any())
             {
-                Reporter.ToUser(eUserMsgKey.ExcelInvalidFieldData);
-                return;
+                DisableSheetNameComboBox();
+                xLoader.Visibility = Visibility.Visible;
+                mExcelOperations.Dispose();
```

Comment:
Why are we calling dispose in between

Reply:
Dispose is used to reinitialize the workBook instance back to 'null'. This way when you switch to a different excel. workBook is null and the code will create an instance of it. It is sort of a Singleton to avoid reading the whole Excel multiple times.
especially when you are toggling the dropdown (the previous implementation recreated the workBook instance by reading the whole excel every time you toggle the drop down).